### PR TITLE
Made JsonArray implement the List interface.

### PR DESCRIPTION
**Pull request by inder123**

### CODE REVIEW

**Comment by inder123** [comment]
@swankjesse @JakeWharton Is this a good idea (to implement List instead of Iterable for JsonArray)? Bloats the API a little bit, but useful nonetheless.

**Comment by swankjesse** [comment]
It isn't a binary-compatible change.

**Comment by swankjesse** [comment]
Don't forget equals().

**Comment by swankjesse** [comment]
It don't feel strongly either way.

If we make this change we will need to figure out the binary compatibility problem.

Also if we make this change we should change JsonObject to implement Map for symmetry.

File: gson/src/main/java/com/google/gson/JsonArray.java [hunk 1]
```diff
@@ -30,7 +32,7 @@
  * @author Inderjeet Singh
  * @author Joel Leitch
  */
-public final class JsonArray extends JsonElement implements Iterable<JsonElement> {
+public final class JsonArray extends JsonElement implements List<JsonElement> {
```
Comment (JakeWharton) [inline gson/src/main/java/com/google/gson/JsonArray.java]: You'll also want the `RandomAccess` marker interface.

**Comment by inder123** [comment]
I agree we should keep binary compatibility. Closing the pull request.

**Pull request closed without merging**